src/Private/Console.ps1

```diff
@@ -0,0 +1,987 @@
+
+.DESCRIPTION
+	This function checks if a key press matches a specified character and modifier combination. It supports detecting Control key presses on all platforms and Shift key presses on Unix systems.
+
+.PARAMETER Key
+	Optional. Specifies the key to test. If not provided, the function retrieves the key using `Get-PodeConsoleKey`.
+
+.PARAMETER Character
+	Mandatory. Specifies the character to test against the key press.
+
+.EXAMPLE
+	Test-PodeKeyPressed -Character 'C'
+
+	Checks if the Control+C combination is pressed.
+
+.NOTES
+	This function is intended for use in scenarios where Pode's console input is enabled.
+#>
+function Test-PodeKeyPressed {
+    param(
+        [Parameter()]
+        $Key = $null,
+
+        [Parameter(Mandatory = $true)]
+        [string]
+        $Character
+    )
+
+    # If console input is disabled, return false
+    if ($PodeContext.Server.Console.DisableConsoleInput) {
+        return $false
+    }
+
+    # Retrieve the key if not provided
+    if ($null -eq $Key) {
+        $Key = Get-PodeConsoleKey
+    }
+
+    # Test the key press against the character and modifiers
+    return (($null -ne $Key) -and ($Key.Key -ieq $Character) -and
+        (($Key.Modifiers -band [ConsoleModifiers]::Control) -or ((Test-PodeIsUnix) -and ($Key.Modifiers -band [ConsoleModifiers]::Shift))))
+}
+
+<#
+.SYNOPSIS
+	Gets the next key press from the Pode console.
+
+.DESCRIPTION
+	This function checks if a key is available in the console input buffer and retrieves it. If the console input is redirected or no key is available, the function returns `$null`.
+
+.EXAMPLE
+	Get-PodeConsoleKey
+
+	Retrieves the next key press from the Pode console input buffer.
+
+.NOTES
+	This function is useful for scenarios requiring real-time console key handling.
+#>
+function Get-PodeConsoleKey {
+    if ([Console]::IsInputRedirected -or ![Console]::KeyAvailable) {
+        return $null
+    }
+
+    return [Console]::ReadKey($true)
+}
+
+<#
+.SYNOPSIS
+    Processes console actions and cancellation token triggers for the Pode server using a centralized key mapping.
+
+.DESCRIPTION
+    The `Invoke-PodeConsoleAction` function uses a hashtable to define and centralize key mappings,
+    allowing for easier updates and a cleaner implementation.
+
+.NOTES
+    This function is part of Pode's internal utilities and may change in future releases.
+
+.EXAMPLE
+    Invoke-PodeConsoleAction
+
+    Processes the next key press or cancellation token to execute the corresponding server action.
+#>
+function Invoke-PodeConsoleAction {
+
+    # Centralized key mapping
+    $KeyBindings = $PodeContext.Server.Console.KeyBindings
+
+    $serverState = Get-PodeServerState
+
+    # Get the next key press if console input is enabled
+    if (!$PodeContext.Server.Console.DisableConsoleInput) {
+        $Key = Get-PodeConsoleKey
+    }
+
+    # Browser action
+    if (Test-PodeKeyPressed -Key $Key -Character $KeyBindings.Browser) {
+        Clear-PodeKeyPressed
```

Comment:
It might be worth adding a `-Clear` switch to `Get-PodeConsoleKey`, and if supplied it calls `Clear-PodeKeyPressed` instead - would clear up all of these numerous `Clear-PodeKeyPressed` calls

Reply:
I remember I did like that but I had some issue 
let me try again maybe now after all these changes maybe works

Reply:
it works I'm removing all these call